src/lsm/compaction.zig

```diff
@@ -453,6 +486,9 @@ pub fn CompactionType(
             assert(compaction.callback != null);
             assert(compaction.io_pending == 0);
             assert(!compaction.merge_done);
+            assert(!compaction.move_table());
+
+            const grid_reservation = compaction.grid_reservation.?;
```

Comment:
@kprotty What is the reason for caching this variable here? Line length?

Per TigerStyle, it's better to avoid introducing a variable into the scope like this, because then there's more chance that something is misused. We've had bugs like this before.

Instead, shrink the scope, by continuing to use `grid_reservation.?` inline as before.

Reply:
I moved it to be a checked access at the start to show that its active for the entire function and that it's immutable. I'll change it back.